agent/local/state.go

```diff
@@ -1331,3 +1342,24 @@ func (l *State) notifyIfAliased(serviceID structs.ServiceID) {
 		}
 	}
 }
+
+// ACLAccessorID takes an ACL secret token and retrives identity metadata on the
+// token, returning just the accessorID so we can log it here. In the future we
+// may wish to return more of the fields on structs.ACLIdentity to log out more
+// debug info when ACLs are denied in state.go.
+func (l *State) ACLAccessorID(token string) string {
+	req := structs.ACLRequest{
+		Datacenter:   l.config.Datacenter,
+		WriteRequest: structs.WriteRequest{Token: token},
+	}
+	var reply structs.ACLIdentity
+	if err := l.Delegate.RPC("ACL.ResolveIdentityFromToken", &req, &reply); err != nil {
```

Comment:
I think instead of using an RPC here it would be better to modify the local states delegate interface to include your `ResolveIdentityFromToken` method and use that.

That will enable the local state to use the ACLResolver and its cache instead of always issuing RPCs. It has the added benefit of not requiring any thought about what happens during upgrades when some servers would be unable to handle this RPC.

Reply:
will do!